.github/workflows/scenarios-permian.yml

```diff
@@ -0,0 +1,120 @@
+# Run all kickstart tests for all active branches in Permian
+name: Daily run in Permian
+on:
+  workflow_dispatch:
+
+permissions:
+  contents: read
+
+jobs:
+  scenario:
+    name: Scenario in Permian
+    runs-on: [self-hosted, kstest-permian]
+    strategy:
+      matrix:
+        scenario: [daily-iso, rhel8, rhel9]
+      fail-fast: false
+
+    # these settings depend on the infrastructure; on upshift ocp-master-xxl they take about 4 hours
+    timeout-minutes: 480
+    env:
+      TEST_JOBS: 16
+      GITHUB_TOKEN: /home/github/github-token
+
+    steps:
+      # self-hosted runners don't do this automatically; also useful to keep stuff around for debugging
+      # need to run sudo as the launch script and the container create root/other user owned files
+      - name: Clean up previous run
+        run: |
+          sudo podman ps -q --all --filter='ancestor=kstest-runner' | xargs -tr sudo podman rm -f
+          sudo podman volume rm --all || true
+          sudo rm -rf *
+
+      - name: Clone repository
+        uses: actions/checkout@v2
+        with:
+          path: kickstart-tests
+
+      - name: Generate test cases
+        working-directory: ./kickstart-tests
+        run: scripts/generate-testcases.py -t ./testlib/test_cases/kstest-template.tc.yaml.j2 . -o ./testlib/test_cases
+
+      - name: Clone Permian repositories
+        run: |
+          git clone https://github.com/rhinstaller/permian.git
+          git clone https://github.com/rhinstaller/tclib.git
+
+      # use the latest official packages for the nightly runs
+      - name: Clean up squid cache
+        run: sudo containers/squid.sh clean
+        working-directory: ./kickstart-tests
+
+      - name: Ensure http proxy is running
+        run: sudo containers/squid.sh start
+        working-directory: ./kickstart-tests
+
+      - name: Set platform from scenario
+        id: platform_from_scenario
+        run: |
+          set -eux
+          if [ "${{ matrix.scenario }}" == "daily-iso" ] || [ "${{ matrix.scenario }}" == "minimal" ]; then
+            echo "::set-output name=platform::fedora_rawhide"
+          elif [ "${{ matrix.scenario }}" == "rhel8" ]; then
+            echo "::set-output name=platform::rhel8"
+          elif [ "${{ matrix.scenario }}" == "rhel9" ]; then
+            echo "::set-output name=platform::rhel9"
```

Comment:
I'm thinking if instead of these if's these shouldn't be specified by the matrix. That would simplify readability and future changes.

Reply:
We may consider this as separate issue, but scenario (eg daily-iso / rawhide - configures the tested boot.iso) and platform (eg fedora_rawhide - configures the platform specific fragments ... --platform argument) seem to me as clearly different things.